lib/bot/bot.js

```diff
@@ -182,6 +183,45 @@ externals.Bot = class {
       parsedMessage.message.commandPrefix || _.toUpper(this.getId()) ===
       parsedMessage.message.commandPrefix;
     const isSelf = parsedMessage.user === _.toUpper(this.getId());
+    const options = {
+      botToken: this.config.botToken,
+      agent: this.getHttpAgent(),
+      user: parsedMessage.user,
+    };
+    let isAllowedUser = false;
```

Comment:
consider extracting both `fetchUserInfo` and `ActiveDirectory` calls into a separate function outside of  `handleMessage`.